Tests/SonarScanner.MSBuild.Test/RollForwardTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarScanner.MSBuild.Test
+{
+    [TestClass]
+    public class RollForwardTests
+    {
+        [TestMethod]
+        public void MustRollForward()
+        {
+            var scannerCsProjRelativePath = Path.Combine("src", "SonarScanner.MSBuild", "SonarScanner.MSBuild.csproj");
```

Comment:
What's the reason for checking the source project file rather than the output 
`SonarScanner.MSBuild.runtimeconfig.json` that will ship to customers?

I guess one problem is locating all of the output files to check, since there will be several of them. There will be one copied locally to the test bin folder, but only for one target framework.

Reply:
For simplicity, we can use `TestContext.DeploymentDirectory` + `SonarScanner.MSBuild.runtimeconfig.json` to assert that TFM is `net5.0` and `rollForward` is `LatestMajor`.

That will break in case we remove or change the config. And will ensure that the latest TFM we ship gets the property right.

Reply:
in the end, after doing [this change](https://github.com/SonarSource/sonar-scanner-msbuild/pull/1114/commits/9213a0b734f880cbfbb146db74db5dacf53d5dee) the test still had a different behavior on the CI ([See failure](https://dev.azure.com/sonarsource/DotNetTeam%20Project/_build/results?buildId=39961&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=831571d4-a404-5634-b803-0eeee1b877b9&l=927)).

To keep things simple, rather than understanding why that runtime.config didn't get copied to the folder on the CI for the release run (when running the Unit Test), I chose to verify the folder in `sonar-scanner-msbuild\build` which are used to create the ZIP archives inside an integration test.

I removed the unit test and replaced it with an integration test.